src/main/scala/midas/models/dram/Util.scala

```diff
@@ -144,17 +145,19 @@ class DynamicLatencyPipe[T <: Data] (
   }
 
   val latencies = Reg(Vec(entries, UInt(countBits.W)))
-  val pending = RegInit(VecInit(Seq.fill(entries)(false.B)))
-  latencies.zip(pending) foreach { case (lat, pending) =>
-    when (lat === io.tCycle) { pending := false.B }
-  }
+  val pendingReg = RegInit(VecInit(Seq.fill(entries)(false.B)))
+  val done = Vec(latencies.zip(pendingReg) map { case (lat, pendingReg) =>
```

Comment:
I would prefer you name the `pendingReg` argument in the case statement something else instead of shadowing the original variable.

Reply:
Good call.